packages/gatsby/src/query/file-parser.ts

```diff
@@ -74,12 +100,22 @@ Perhaps the variable name has a typo?
 Also note that we are currently unable to use queries defined in files other than the file where the ${usageFunction} is defined. If you're attempting to import the query, please move it into "${file}". If being able to import queries from another file is an important capability for you, we invite your help fixing it.\n`
   )
 
-async function parseToAst(filePath, fileStr, { parentSpan, addError } = {}) {
-  let ast
+async function parseToAst(
+  filePath: string,
+  fileStr: string,
+  {
+    parentSpan,
+    addError,
+  }: {
+    parentSpan?: Span
+    addError?: (opt: any) => void
```

Comment:
Fwiw... Looking at this file, I _think_ that `addError` is not optional. This function is called in one position, and the function that calls it is used in one position, and in that position `addError` is assumed to exist.